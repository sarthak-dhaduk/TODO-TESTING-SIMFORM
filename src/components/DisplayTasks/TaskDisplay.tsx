import React from "react";
import { TaskDisplayProps } from "./TaskDisplay.types";

const TaskDisplay = ({ list, type }: TaskDisplayProps) => {
  return (
    <>
      <ul>
        {list?.map((value, idx) => {
          return (
            <li
              className="flex justify-between items-center mt-8"
              key={value + idx}
            >
              <div
                className={`break-words text-justify flex-1 mx-2 max-w-[80%] text-lg ${
                  type === "completed" ? "text-textComplete" : ""
                }`}
              >
                {value}
              </div>
              <button className="w-9">
                <img src={`/assets/task-${type}.png`} alt="completed" />
              </button>
            </li>
          );
        })}
      </ul>
    </>
  );
};

export default TaskDisplay;
