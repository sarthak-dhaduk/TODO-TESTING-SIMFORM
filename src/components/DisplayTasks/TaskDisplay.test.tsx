import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import TaskDisplay from "./TaskDisplay";

//mock tasks
const taskList = [
  "Begin promotional phase",
  "Read an article",
  "Begin QA for the product",
];

describe("TaskDisplay", () => {
  //testing the ui snapshot
  test("TaskDisplay renders correctly", () => {
    const renderedComponent = renderer
      .create(<TaskDisplay list={taskList} type="pending" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  //renders the tasks
  test("renders a tasks list", () => {
    render(<TaskDisplay list={taskList} type="completed" />);
    const listItemElements = screen.getAllByRole("listitem");
    expect(listItemElements).toHaveLength(taskList.length);
  });
});
