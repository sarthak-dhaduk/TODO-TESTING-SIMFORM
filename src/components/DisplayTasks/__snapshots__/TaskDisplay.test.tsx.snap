// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDisplay TaskDisplay renders correctly 1`] = `
<ul>
  <li
    className="flex justify-between items-center mt-8"
  >
    <div
      className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
    >
      Begin promotional phase
    </div>
    <button
      className="w-9"
    >
      <img
        alt="completed"
        src="/assets/task-pending.png"
      />
    </button>
  </li>
  <li
    className="flex justify-between items-center mt-8"
  >
    <div
      className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
    >
      Read an article
    </div>
    <button
      className="w-9"
    >
      <img
        alt="completed"
        src="/assets/task-pending.png"
      />
    </button>
  </li>
  <li
    className="flex justify-between items-center mt-8"
  >
    <div
      className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
    >
      Begin QA for the product
    </div>
    <button
      className="w-9"
    >
      <img
        alt="completed"
        src="/assets/task-pending.png"
      />
    </button>
  </li>
</ul>
`;
